lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1603,10 +1603,6 @@ public function getEntityState($entity, $assume = null)
             return self::STATE_NEW;
         }
 
-        if ($class->containsForeignIdentifier) {
```

Comment:
Not sure about this, would need to review the side effects on a computer with IDE (on mobile app here)

Reply:
This was moved into UoW tryGetById

Reply:
This was indeed breaking stuff, so I reverted this change to original code